src/moonlink/src/storage/mooncake_table/table_operation_test_utils.rs

```diff
@@ -260,6 +260,7 @@ async fn sync_mooncake_snapshot_and_create_new_by_iceberg_payload(
         skip_file_indices_merge: true,
         skip_data_file_compaction: true,
     }));
+    sync_mooncake_snapshot(table, receiver).await;
```

Comment:
This is a bug in the unit test, which could lead to flaky test, since we're not properly block wait mooncake snapshot completion.